### PR TITLE
Add streaming message receive and send histograms

### PR DESCRIPTION
**Pull request by povilasv**

### CODE REVIEW

**Comment by povilasv** [comment]
I signed it!

File: client_metrics.go [hunk 1]
```diff
@@ -13,13 +13,19 @@ import (
...
-	clientHandledHistogram        *prom.HistogramVec
+	clientStartedCounter             *prom.CounterVec
+	clientHandledCounter             *prom.CounterVec
+	clientStreamMsgReceived          *prom.CounterVec
+	clientStreamMsgSent              *prom.CounterVec
+	clientHandledHistogramEnabled    bool
+	clientHandledHistogramOpts       prom.HistogramOpts
+	clientHandledHistogram           *prom.HistogramVec
+	clientStreamRecvHistogramEnabled bool
+	clientStreamRecvHistogramOpts    prom.HistogramOpts
```
Comment (bwplotka) [inline client_metrics.go]: nit: Can  you add newlines to separate those groups?

File: client_metrics.go [hunk 2]
```diff
@@ -104,6 +136,40 @@ func (m *ClientMetrics) EnableClientHandlingTimeHistogram(opts ...HistogramOptio
 	m.clientHandledHistogramEnabled = true
 }
 
+// EnableClientStreamReceiveTimeHistogram turns on recording of single message receive time of streaming RPCs.
+// Histogram metrics can be very expensive for Prometheus to retain and query.
+func (m *ClientMetrics) EnableClientStreamReceiveTimeHistogram(opts ...HistogramOption) {
+	for _, o := range opts {
+		o(&m.clientHandledHistogramOpts)
```
Comment (bwplotka) [inline client_metrics.go]: Typo? 

```suggestion
		o(&m.clientStreamRecvHistogramOpts)
```

File: client_metrics.go [hunk 3]
```diff
@@ -104,6 +136,40 @@ func (m *ClientMetrics) EnableClientHandlingTimeHistogram(opts ...HistogramOptio
...
+	}
+
+	m.clientStreamRecvHistogramEnabled = true
+}
+
+// EnableClientStreamSendTimeHistogram turns on recording of single message send time of streaming RPCs.
+// Histogram metrics can be very expensive for Prometheus to retain and query.
+func (m *ClientMetrics) EnableClientStreamSendTimeHistogram(opts ...HistogramOption) {
+	for _, o := range opts {
+		o(&m.clientHandledHistogramOpts)
```
Comment (bwplotka) [inline client_metrics.go]:
```suggestion
		o(&m.clientStreamSendHistogramOpts)
```

**Comment by povilasv** [comment]
> So there's good news and bad news.
> 
>  The good news is that everyone that needs to sign a CLA (the pull request submitter and all commit authors) have done so. Everything is all good there.
> 
>  The bad news is that it appears that one or more commits were authored or co-authored by someone other than the pull request submitter. We need to confirm that all authors are ok with their commits being contributed to this project. Please have them confirm that here in the pull request.
> 
> _Note to project maintainer: This is a terminal state, meaning the `cla/google` commit status will not change from this state. It's up to you to confirm consent of all the commit author(s), set the `cla` label to `yes` (if enabled on your project), and then merge this pull request when appropriate._
> 
>  **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fgrpc-ecosystem%2Fgo-grpc-prometheus%2Fpull%2F71) for more info**.

@bwplotka  could you confirm this :D and set cla label to yes

**Comment by bwplotka** [comment]
Lol ;p

**Pull request merged**